h2/src/main/org/h2/index/IndexCondition.java

```diff
@@ -64,7 +66,12 @@ public class IndexCondition {
      */
     public static final int SPATIAL_INTERSECTS = 16;
 
-    private final Column column;
+    /**
+     * Contains a {@link Column} or {@code Column[]} depending on the condition type.
+     * @see #isCompoundColumns()
+     */
+    private final Object column;
```

Comment:
It will be better to use different fields here.

Reply:
Done